### PR TITLE
[Docs]Updated examples for customizing resources

### PR DESCRIPTION
**Pull request by LunarMarathon**
# TL;DR
Closes: [4035](https://github.com/flyteorg/flyte/issues/4035)
Updated the examples to use `with_overrides(requests=Resources(...), limits=Resources(...))` instead of what was defined in the older version of flytekit.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/4035

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by kumare3** [comment]
LGTM, but you have to sign the DCO

**Comment by LunarMarathon** [comment]
Done👍 @kumare3

File: tests/flytekit/unit/core/test_map_task.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import functools
 import typing
 from collections import OrderedDict
+from flytekit.core.resources import Resources
```
Comment (samhita-alla) [inline tests/flytekit/unit/core/test_map_task.py:4]:
```suggestion
from flytekit import Resources
```

Comment (samhita-alla) [reply tests/flytekit/unit/core/test_map_task.py:4]: Can you also sort the imports using isort?

Comment (LunarMarathon) [reply tests/flytekit/unit/core/test_map_task.py]: Done👍

**Comment by samhita-alla** [comment]
@LunarMarathon, can you fix the DCO?

**Comment by LunarMarathon** [comment]
@samhita-alla I'm sorry, I completely forgot about signing off the second commit. I've fixed it.

**Review by pingsutw — APPROVED** [review]

**Review by samhita-alla — APPROVED** [review]
